wordpress-mobile/AztecEditor-iOS
# 492: Style hierarchy

This PR implements a new paragraph style hierarchy.

All the attributes of a paragraph (header level, lists, blockquotes) are now in a hierarchy and they can be nested like in HTML.
The base class of the properties is the `ParagraphProperties` class. This class contains an optional `HTMLRepresentation` that can be used to store the original HTML attributes and element used.

I kept the old properties available but only has getters to allow a smooth transition of the old code, specially the list layout code.

To test:
 - Launch the demo app
 - See if all content shows up correctly
 - Apply toggle of the toolbar and check if they work correctly: Header, Lists, Blockquotes, etc..
